### DISCUSSION TITLE
Define custom name and custom type for foreign key.

### DISCUSSION THREAD

**Discussion by wasya243** [typegoose/typegoose#814] [Q&A] [answered]
Here is the code.

```ts
import { prop, getModelForClass, Ref } from '@typegoose/typegoose';
import mongoose from 'mongoose';

export class Organization {
  @prop({required: true})
  name!: string;

  @prop({
    ref: () => User,
    foreignField: 'organizationId',
    localField: '_id'
  })
  public users: Ref<User>[] = [];
}
export class User {
  @prop({required: true})
  name!: string;

  @prop({
    ref: () => Organization, 
    required: true,
    foreignField: '_id',
    localField: 'organizationId',
    justOne: true
  })
  private organization!: Ref<Organization>;

  @prop({ required: true })
  public organizationId!: string;

}

const OrganizationModel = getModelForClass(Organization, {
  schemaOptions: {
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
  }
});

const UserModel = getModelForClass(User, {
  schemaOptions: {
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
  }
});
async function main() {
  try {
    await mongoose.connect('mongodb://localhost:27017/', {dbName: 'migration-to-typegoose'});

    const organization = await OrganizationModel.create({
      name: 'Acme Inc.'
    });

    const users = await Promise.all([
      UserModel.create({
        name: 'Vasyl',
        organizationId: organization._id.toString()
      }),
      UserModel.create({
        name: 'Misha',
        organizationId: organization._id.toString()
      })
    ]);

    const user = await UserModel.findOne().populate('organization');

    console.log('user', user)

    process.exit(0);
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
}

main();
```

My question is there any way to define one to many into more concise way ('cause here I define both organization and organizationId fields)? Is it possible to define such relationship using only 1 field? Yeap, it's intended that organizationId is a String. It's intended. It's the data I'm currently working on (let's assume migration String -> ObjectId to make things simplier) is not possible.

**Comment by hasezoey** [comment, marked as answer]
because you are only using a simple reference, you could throw away `organization` and just use `organizationId` via `public organizationId!: Ref<Organization, string>;`, assuming `Organization`'s `_id` type is actually a string (unlike in the example where it is implicitly `ObjectId`), otherwise the only other way is to use `ObjectId` for both `organizationId` and `Organization._id`, if that is not possible, then no, there is no way to use less than 2 fields
Note: it *may* work to use differing types (in the case of String and ObjectId) and it will automatically coerce into the same type, but i am not sure about that

**Reply by wasya243** [reply]
Thanks!
